x-pack/plugins/upgrade_assistant/public/application/components/shared/deprecation_badge.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiBadge } from '@elastic/eui';
+
+const i18nTexts = {
+  criticalBadgeLabel: i18n.translate('xpack.upgradeAssistant.deprecationBadge.criticalBadgeLabel', {
+    defaultMessage: 'Critical',
+  }),
+  resolvedBadgeLabel: i18n.translate('xpack.upgradeAssistant.deprecationBadge.resolvedBadgeLabel', {
+    defaultMessage: 'Resolved',
+  }),
+  warningBadgeLabel: i18n.translate('xpack.upgradeAssistant.deprecationBadge.warningBadgeLabel', {
+    defaultMessage: 'Warning',
+  }),
+};
+
+interface Props {
+  isCritical: boolean;
+  isResolved?: boolean;
+}
+
+export const DeprecationBadge: FunctionComponent<Props> = ({ isCritical, isResolved }) => {
```

Comment:
I would've liked to pass `deprecation.warning` here but the api that returns the `ES deprecations` is returning `isCritical` only. In a following PR we could refactor this to return the level directly 🔍

Reply:
I've created https://github.com/elastic/kibana/issues/112097 to do this work 🎸